mobile/src/components/LanguageButtons.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useState } from 'react';
+import { Text, StyleSheet, View, TouchableOpacity } from 'react-native';
+import { Colors } from '../styles/Colors';
+import { TextStyles } from '../styles/TextStyles';
+import { Language, Languages } from '../models/Languages';
+
+// button labels and callback function for passing key of active button to parent
+type LanguageButtonsProps = { langs: Language[], defaultActive: string, onBtnChange };
+
+/**
+ * Renders buttons on language selector
+ */
+export const LanguageButtons: React.FC<LanguageButtonsProps> = ({ langs, defaultActive, onBtnChange }) => {
+  const [activeButton, setActiveButton] = useState(defaultActive);
+
+  return (
+    <View style={styles.container}>
+      {langs.map((res, idx) => {
+        return (
+          <View key={res} style={{ width: '33.33%' }}>
+            <TouchableOpacity
+              style={[styles.button,
+                activeButton === res ? styles.buttonActive : styles.buttonInactive,
+                (idx % 3) === 0 && styles.leftBtn,
+                (idx === langs.length - 1 || (idx % 3 === 2)) && styles.rightBtn,
+              ]}
+              onPress={() => {
+                setActiveButton(res);
+                onBtnChange(res);
+              }}
+            >
+              <Text style={[activeButton === res ? styles.buttonTextActive : styles.buttonTextInactive, TextStyles.caption2]}>
+                {Languages[res]}
+              </Text>
+            </TouchableOpacity>
+          </View>
```

Comment:
Nice, so you set a max of 3 languages per row and then style the borders appropriately? That sounds good. Let's maybe make 3 a constant though, add a `const LANGS_PER_ROW = 3` outside of the component, just to make the code a little more readable.

Also I would define the styles on line 19 (inside the map(), outside of the return statement) rather than inline it in the JSX, it's a little hard to read.

Reply:
Also nit: `res` should maybe be called `lang`